src/cpp/session/modules/SessionCopilot.cpp

```diff
@@ -1227,6 +1266,75 @@ void onBackgroundProcessing(bool isIdle)
             }
             continue;
          }
+         else if (method == "window/showMessageRequest")
+         {
+            std::string message = "undetermined";
+            std::string type = "Error"; // assume the worst
+
+            json::Value paramsJson = responseJson["params"];
+            if (paramsJson.isObject())
+            {
+               json::Object params = paramsJson.getObject();
+               json::Value messageJson = params["message"];
+               json::Value typeJson = params["type"];
+
+               if (messageJson.isString())
+               {
+                  message = messageJson.getString();
+               }
+               if (typeJson.isInt())
+               {
+                  int typeInt = typeJson.getInt();
+                  if (typeInt == 1)
+                     type = "Error";
+                  else if (typeInt == 2)
+                     type = "Warning";
+                  else if (typeInt == 3)
+                     type = "Info";
+                  else if (typeInt == 4)
+                     type = "Log";
+                  else if (typeInt == 5)
+                     type = "Debug";
+               }
+            }
+
+            if (boost::iequals(type, "Error") ||
+                boost::iequals(type, "Warning") ||
+                boost::iequals(type, "Info"))
+            {
+               std::string title = "GitHub Copilot";
+               module_context::showErrorMessage(title, message);
```

Comment:
Could we use the more generic 'showDialogEvent' here? Or add as separate `showMessage` client event that also accepts the message severity?

Reply:
We could, but I'm not going to bother at this time. If we start seeing messages at a non-error level we can reevaluate.